miso-web/src/main/webapp/scripts/warning_completion.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2012. The Genome Analysis Centre, Norwich, UK
+ * MISO project contacts: Robert Davey @ TGAC
+ * *********************************************************************
+ *
+ * This file is part of MISO.
+ *
+ * MISO is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MISO is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+//TODO: add consent revoked errors
+
+WarningTarget.completion = {
+    tableWarnings: function(data, type, full){
+      var warnings = [];
+      pool = full.pool;
+      if(pool.duplicateIndices){
```

Comment:
A lot of this code (if data.possibilityX warn x; if data.possibiltyY warn y; ...) is duplicated across the various WarningTarget files. Can it be abstracted into a method?